test/city-council-district/city-council-district.e2e-spec.ts

```diff
@@ -104,4 +107,107 @@ describe("City Council District e2e", () => {
+      expect(() =>
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        ),
+      ).not.toThrow();
+      const parsedBody =
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        );
+      expect(parsedBody.limit).toBe(20);
+      expect(parsedBody.offset).toBe(0);
+    });
+
+    it("should 200 amd return all capital projects for a city council district with user specifyied offset and limit", async () => {
+      const limit = 10;
+      const offset = 3;
+      const response = await request(app.getHttpServer())
+        .get(
+          `/city-council-districts/${mock.id}/capital-projects?limit=${limit}&offset=${offset}`,
+        )
+        .expect(200);
+      expect(() =>
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        ),
+      ).not.toThrow();
+      const parsedBody =
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        );
+      expect(parsedBody.limit).toBe(limit);
+      expect(parsedBody.offset).toBe(offset);
+    });
+
+    it("should 400 and when finding by an invalid id", async () => {
+      const invalidId = "MN";
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${invalidId}/capital-projects`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 400 and when finding by an invalid limit", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?limit=b4d"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 400 and when finding by an 'too-high' limit", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?limit=25"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+    it("should 400 and when finding by an 'too-low' limit", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?limit=0"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 400 and when finding by an invalid offset", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?offset=b4d"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 404 and when finding by an invalid id", async () => {
+      const missingId = "102";
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${missingId}/capital-projects`)
+        .expect(400);
```

Comment:
These assertions do not match the description. The description indicates that the test should check for a missing id. However, the 400 and Bad Request assertions are for cases of "invalid" ids.